src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs

```diff
@@ -9,99 +9,99 @@
-            events.OnChallenge = OnChallengeAsync;
-
-            return events;
-        }
-
-        private async Task OnMessageReceivedAsync(MessageReceivedContext context)
-        {
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodBegin, nameof(OnMessageReceivedAsync)));
-
-            // Place a breakpoint here and examine the bearer token (context.Request.Headers.HeaderAuthorization / context.Request.Headers["Authorization"])
-            // Use https://jwt.ms to decode the token and observe claims
-            await s_onMessageReceived(context).ConfigureAwait(false);
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodEnd, nameof(OnMessageReceivedAsync)));
-        }
-
-        private async Task OnAuthenticationFailedAsync(AuthenticationFailedContext context)
-        {
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodBegin, nameof(OnAuthenticationFailedAsync)));
-
-            // Place a breakpoint here and examine context.Exception
-            await s_onAuthenticationFailed(context).ConfigureAwait(false);
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodEnd, nameof(OnAuthenticationFailedAsync)));
-        }
-
-        private async Task OnTokenValidatedAsync(TokenValidatedContext context)
-        {
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodBegin, nameof(OnTokenValidatedAsync)));
-            await s_onTokenValidated(context).ConfigureAwait(false);
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodEnd, nameof(OnTokenValidatedAsync)));
-        }
-
-        private async Task OnChallengeAsync(JwtBearerChallengeContext context)
-        {
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodBegin, nameof(OnChallengeAsync)));
-            await s_onChallenge(context).ConfigureAwait(false);
-            _logger.LogDebug(string.Format(CultureInfo.InvariantCulture, LogMessages.MethodEnd, nameof(OnChallengeAsync)));
-        }
-    }
+	/// <summary>
+	/// Diagnostics for the JwtBearer middleware (used in web APIs).
+	/// </summary>
+	public class JwtBearerMiddlewareDiagnostics : IJwtBearerMiddlewareDiagnostics
+	{
+		private readonly ILogger _logger;
+
+		/// <summary>
+		/// Constructor for a <see cref="JwtBearerMiddlewareDiagnostics"/>. This constructor
+		/// is used by dependency injection.
+		/// </summary>
+		/// <param name="logger">Logger.</param>
+		public JwtBearerMiddlewareDiagnostics(ILogger<JwtBearerMiddlewareDiagnostics> logger)
+		{
+			_logger = logger;
+		}
+
+		/// <summary>
+		/// Invoked if exceptions are thrown during request processing. The exceptions will be re-thrown after this event unless suppressed.
+		/// </summary>
+		private Func<AuthenticationFailedContext, Task> s_onAuthenticationFailed = null!;
```

Comment:
This naming convention doesn't match the rest of the codebase. Historically, s_ meant static, which makes this naming a bit misleading.